prime-dotnet-webapi/ViewModels/AutoMapping.cs

```diff
@@ -114,5 +119,29 @@ public AutoMapping()
         CreateMap<PlrProvider, PlrProvider>()
             .ForMember(dest => dest.Id, opt => opt.Ignore())
             .ForMember(dest => dest.Ipc, opt => opt.Ignore());
+
+        // ProjectTo does not work for mapping expertise array, use workaround below and call mapper.Map
+        CreateMap<PlrProvider, PlrViewModel>()
```

Comment:
For this instead of having the complicated mapping, we can go into the PlrProvider model and explicitly  set foreign key references to the new lookups you just made.

[ForeignKey("PlrRoleType")]
public string ProviderRoleType { get; set; }

[JsonIgnore]
public PlrRoleType PlrRoleType { get; set; }

Not sure if you need the JsonIgnore, but this will simplify the mapping not to need DB calls in the mapper. And helps describe the relationship in the models. Which is better than hiding this relationship in the mapper.

The only thing we have to watch for is if there is data outside that defined in the lookups. Though the mapper would fail in that case too, just the way above would fail on data insertion to the DB

Reply:
Usually we use our ORM to generate these FK relationships but like everything we can explicitly describe them since we don't want to change the names of the columns from an external source.

Reply:
> Usually we use our ORM to generate these FK relationships but like everything we can explicitly describe them since we don't want to change the names of the columns from an external source.

This is the reason I am hesitated to use foreignkey since these PLR and lookup data are from external source and could change over time. Do we really want to maintain and keep them in sync?

Reply:
Did we generate these lookup table values or are they defined by PLR?

Reply:
Looking at the ticket it seems these were given to us from PLR, So I think it's safe to say if their data breaks these definitions then they are on the hook for that. I'd have to look at how we do insertion, but realistically if should just not insert rows with no foreign key constraint. In my mind this isn't a case we should try and make a work around in our code for, unless we have been explicitly told that PLR change these type def.

Reply:
One thing to note is 
``` .FirstOrDefault().Name ``` is going to throw an error when code is not in the lookup on 126/127 so you will have to handle the case.

You can do this by defining a default or use ?.

Reply:
Also was just talking to Alan and he is going to reach out to PLR as it may be that they would notify us when they are adding new values so that having the foreign key is a reasonable solution.

Reply:
> One thing to note is
> `.FirstOrDefault().Name` is going to throw an error when code is not in the lookup on 126/127 so you will have to handle the case.
> 
> You can do this by defining a default or use ?.

Tested the null case, and surprisingly no exception is thrown. might be due to it's a IQueryable?

Reply:
> Also was just talking to Alan and he is going to reach out to PLR as it may be that they would notify us when they are adding new values so that having the foreign key is a reasonable solution.

@hogmann Foreignkey is added to PlrProvider model for RoleType and StatusReason. However Expertise is an array property deserialized from JSON string in db so could not be mapped using foreignkey.

Reply:
> Also was just talking to Alan and he is going to reach out to PLR as it may be that they would notify us when they are adding new values so that having the foreign key is a reasonable solution.

Hi @hogmann , per feedback from @itchison  we'd like to always accept PLR messages so FK might cause rejection if the lookup table is not updated in a timely manner or data out of sync. I've reverted the code. Can you take another look?